File: Packets/packets/PlayerReady.cs
using System;
using System.Text;
using Game.Enums;

public class PlayerReady : Packet<PacketType>
{
    public string PlayerId {get; private set;}
    public byte Healt {get; private set;}
    public byte Weapon {get; private set;}
    public byte Damping {get; private set;}

    public PlayerReady(byte[] bytes) : base(bytes)
    {
    }

    public PlayerReady(string playerId, byte healt, WeaponType weapon, byte damping, PacketAction action) : base(PacketType.PlayerReady, action)
    {
        this.PlayerId = playerId;
        this.Healt = healt;
        this.Weapon = (byte)weapon;
        this.Damping = damping;
    }

    protected override void ReadPayload(byte[] bytes, byte offset)
    {
        PlayerId = Encoding.UTF8.GetString(bytes, offset, 8);
        Healt = bytes[offset + 8];
        Weapon = bytes[offset + 9];
        Damping = bytes[offset + 10];

    }

    protected override void WritePayload(byte[] bytes, byte offset)
    {
        Array.Copy(Encoding.UTF8.GetBytes(PlayerId), 0, bytes, offset, 8);
        bytes[offset + 8] = Healt;
        bytes[offset + 9] = Weapon;
        bytes[offset + 10] = Damping;


    }

    public override string ToString()
    {
        return $"AppStarted{{playerId='{PlayerId}'}}";
    }
}

File: src/main/java/net/nerfatg/proxy/packet/packets/PlayerReady.java
package net.nerfatg.proxy.packet.packets;

import net.nerfatg.proxy.packet.Packet;
import net.nerfatg.proxy.packet.PacketAction;
import net.nerfatg.proxy.packet.PacketType;

import java.nio.BufferOverflowException;
import java.nio.ByteBuffer;
import java.nio.charset.StandardCharsets;

public class PlayerReady extends Packet {
    private String playerId;
    private byte health;
    private byte weapon;
    private byte damping;

    public PlayerReady(ByteBuffer buffer) throws BufferOverflowException {
        super(buffer);
    }

    public PlayerReady(String playerId, byte health, byte weapon, byte damping, PacketAction action) {
        super(PacketType.PlayerReady, action);
        this.playerId = playerId;
        this.health = health;
        this.weapon = weapon;
        this.damping = damping;
    }

    @Override
    public void readPayload(ByteBuffer buffer, int offset) throws BufferOverflowException {
        buffer.position(offset);
        byte[] playerIdBytes = new byte[8];
        buffer.get(playerIdBytes);
        this.playerId = new String(playerIdBytes, StandardCharsets.UTF_8);
        this.health = buffer.get();
        this.weapon = buffer.get();
        this.damping = buffer.get();
    }

    @Override
    public void writePayload(ByteBuffer buffer, int offset) throws BufferOverflowException {
        buffer.position(offset);
        
        // Write playerId (8 bytes fixed)
        byte[] originalPlayerIdBytes = playerId.getBytes(StandardCharsets.UTF_8);
        buffer.put(originalPlayerIdBytes, 0, Math.min(originalPlayerIdBytes.length, 8));
        if (originalPlayerIdBytes.length < 8) {
            buffer.put(new byte[8 - originalPlayerIdBytes.length]); // pad with zeros
        }
        
        buffer.put(health);
        buffer.put(weapon);
        buffer.put(damping);
    }

    public String getPlayerId() { return playerId; }
    public byte getHealth() { return health; }
    public byte getWeapon() { return weapon; }
    public byte getDamping() { return damping; }

    @Override
    public String toString() {
        return String.format("PlayerReady{playerId='%s', health=%d, weapon=%d, damping=%d}", playerId, health, weapon, damping);
    }
}